jerry-core/ecma/operations/ecma-regexp-object.h

```diff
@@ -29,26 +29,30 @@
  * @{
  */
 
-#define RE_EXECUTE_RECURSION_LIMIT  1000  /* Limit of RegExp executor recursion depth */
-#define RE_EXECUTE_MATCH_LIMIT      10000 /* Limit of RegExp execetur matching steps */
+/**
+ * Limit of RegExp executor recursion depth
+ */
+#define RE_EXECUTE_RECURSION_LIMIT  1000
```

Comment:
@ruben-ayrapetyan, a simple example is using groups. Each group usage increase recursion limit by one.
Maximum recursion of `/a/` is one, `/(a)/` is two, `/((a))/` is three. Iteration of groups also increase the recursion depth. Maximum recursion of `/(a){100}/` is 101. So it can restrict the engine, but not in specification-defined semantics. You cannot run too big or complex RegExp. In the current implementation the engine should not lead to infinite loop for any expressions. This is only a sanity check now.